src/Form/ContactMessageForm.php

```diff
@@ -69,6 +69,12 @@ public function save(array $form, FormStateInterface $form_state): void {
     $message = $this->entity;
     /** @var \Drupal\contact\ContactFormInterface $contact_form */
     $contact_form = $message->getContactForm();
+    // Redirect path field is not filled by user.
+    if (!$contact_form->getRedirectPath()) {
+      // Remove front page redirect, it's a known core issue.
```

Comment:
This behaviour is only for corporate (business logic) so no use of pointing out the core issue, best you remove the comment here altogether

Reply:
Yes I implemented this but the test is not fine yet. I'll have a look in the upcoming days.